### ISSUE TITLE
Clippy warns about ineffective identity_op

### ISSUE THREAD

**Issue by dtolnay** [serde-rs/serde#983]
```rust
#[macro_use]
extern crate serde_derive;

#[derive(Serialize)]
struct S {
    a: u8,
}

fn main() {}
```

```
warning: the operation is ineffective. Consider reducing it to `Serialize`
 --> src/main.rs:4:10
  |
4 | #[derive(Serialize)]
  |          ^^^^^^^^^
  |
  = note: #[warn(identity_op)] on by default
  = help: for further information visit https://github.com/Manishearth/rust-clippy/wiki#identity_op
```

We have seen this before but it was fixed in Clippy I think. #395 has a workaround.

**Comment by dtolnay** [comment]
Also affects single_match_else: https://github.com/Manishearth/rust-clippy/issues/1877

**Comment by dtolnay** [comment]
Fixed as of clippy 0.0.144.
